assemblers/org.osbuild.qemu

```diff
@@ -52,9 +52,33 @@ SCHEMA = """
     }
   },
   "format": {
-    "description": "Image file format to use",
-    "type": "string",
-    "enum": ["raw", "raw.xz", "qcow2", "vdi", "vmdk", "vpc", "vhdx"]
+    "oneOf": [
```

Comment:
Overloading the format (string vs. object) is going to make it difficult for strictly typed languages (eg. Go) to handle this. I'd just add new fields next to format.

Reply:
I assume the old form is just there for backwards compatibility and strictly typed languages will just only use the object form.

Reply:
I feel the same. Also, backwards compatibility with the "flat schema" would be a bit of a mess.

Reply:
Ah, yes, that makes sense.